### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by braincee**
_**In this pull request, the following requirements were implemented:**_

- I created a class Nameable.
- I implemented a method called correct_name that will raise a NotImplementedError.
- I turned your Person class to Nameable
- I made sure that my Person class inherits from Nameable
I made sure that this class has a method correct_name implemented. It should simply return the name attribute.

-  Prepare base Decorator

- I made sure that it inherits from Nameable.
- In the constructor, I assign a nameable object from params to an instance variable.
- I implemented the correct_name method that returns the result of the correct_name method of the @nameable.

-  Prepare CapitalizeDecorator and TrimmerDecorator

- For the CapitalizeDecorator:
- I created a class that inherits from the base Decorator class.
- implemented a method correct_name that capitalises the output of @nameable.correct_name.
- For the TrimmerDecorator:
- I created a class that inherits from the base Decorator class.
- I implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable = nameable.new())
+    super()
```
Comment (julie-ify) [inline base_decorator.rb]: - [ ] I received the attached error when I ran your program, and the error was about the default parameter you passed to the initialize method as an instance of the `Nameable` class, which is not supposed. Kindly remove the default value and pass `nameable` without the default value.

- [ ] Also, there is no use of the `super()` call you to have here too. Kindly remove it.

chrome_IF8oZB8Wiu

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @braincee,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def correct_name
+    @name
+  end
```
Comment (julie-ify) [inline person.rb:28]: - [ ] **[OPTIONAL]** I forgot to mention in the first review that you should not make this method private. I got an error saying that I'm trying to access the private method `correct_name` in the person class. So please kindly make this method `public` before your next milestone.

**Review by julie-ify — APPROVED** [review]
Hi @braincee, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by braincee** [comment]
Thanks very much, @julie-ify 

**Note: Please I have implemented all the required and optional changes.**

thank-you-thank-you-thank-you

**Pull request merged**